From b966dccd427fba8f70e77cb1bed10e8f7393b3e3 Mon Sep 17 00:00:00 2001
From: "djm@openbsd.org" <djm@openbsd.org>
Date: Sun, 27 Oct 2024 02:06:59 +0000
Subject: upstream: explicitly include endian.h

OpenBSD-Commit-ID: 13511fdef7535bdbc35b644c90090013da43a318

Origin: upstream, https://anongit.mindrot.org/openssh.git/commit/?h=V_9_9&id=fe8d28a7ebbaa35cfc04a21263627f05c237e460
Last-Update: 2024-10-27

Patch-Name: mlkem768x25519-big-endian-2.patch
---
 kexmlkem768x25519.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/kexmlkem768x25519.c b/kexmlkem768x25519.c
index 679446e97..2b5d39608 100644
--- a/kexmlkem768x25519.c
+++ b/kexmlkem768x25519.c
@@ -1,4 +1,4 @@
-/* $OpenBSD: kexmlkem768x25519.c,v 1.1 2024/09/02 12:13:56 djm Exp $ */
+/* $OpenBSD: kexmlkem768x25519.c,v 1.2 2024/10/27 02:06:59 djm Exp $ */
 /*
  * Copyright (c) 2023 Markus Friedl.  All rights reserved.
  *
@@ -34,6 +34,9 @@
 #include <stdbool.h>
 #include <string.h>
 #include <signal.h>
+#ifdef HAVE_ENDIAN_H
+# include <endian.h>
+#endif
 
 #include "sshkey.h"
 #include "kex.h"
